src/config.c

```diff
@@ -1061,6 +1100,16 @@ struct rewriteConfigState *rewriteConfigReadOldFile(char *path) {
             continue;
         }
 
+        /* If the config file contains a configuration that is no longer present
+         * in the configs array: remove it from the .conf file. Use case right now
+         * is module with configurations unloaded after a rewrite with the module configs. */
+        if (!server.sentinel_mode && !lookupConfig(argv[0])) {
+            linenum--;
+            sdsfreesplitres(argv, argc);
+            sdsfree(line);
+            continue;
+        }
```

Comment:
This seems good. Originally I suggested commenting out these lines instead of completely removing them. WDYT?

Reply:
I don't necessarily feel particularly strongly either way. It is also a larger question for the future I think, as there may be future use cases where configs only exist for a certain period of runtime. To me it feels somewhat more intuitive that they would be removed, but again I don't feel all that strongly.

Reply:
I guess even changing this in the future won't be a breaking change, and we do want to avoid a situation where the re-written config keeps growing because we're inserting comment lines. So I'm fine with leaving this as is.

Reply:
i gave it some thought and here are a few points i have:
[1] this is probably not a common issue (doing CONFIG REWRITE, then MODULE UNLOAD and CONFIG REWRITE again).

[2] removing a line with a config could mean that it leaves behind some comment documenting that line.
i think this is bad.
e.g.
```
# we disable this config because bla bla....
some_config no
```

so i think we should comment these lines, but it can be maybe done in a followup PR, not a mandatory task for this one.